### DISCUSSION TITLE
How to change type on allready inserted elements in canvas.

### DISCUSSION THREAD

**Discussion by ikenderham** [GrapesJS/grapesjs#4794] [Q&A]
Hello everyone. 

I need a solution to change type on default components to "section". I figured out how to change type on new elements, but all the allready inserted, is still "default".

//const wrapper = editor.DomComponents.getWrapper();
//wrapper.find('.row.segment').forEach(comp => comp.addAttributes({ type: 'sektion' }));

please can anyone help? Just to save time and not have to remake all my designs to get type "sektion"

**Comment by artf** [comment]
You need to update the property of the component and not the attribute `comp.set({ type: 'sektion' })`

**Reply by ikenderham** [reply]
I tried with 

const wrapper = editor.DomComponents.getWrapper();
wrapper.find('.row.segment').forEach(comp => comp.set({ type: 'sektion' }));

But its not changing the types.

**Reply by ikenderham** [reply]
I tried with

const wrapper = editor.DomComponents.getWrapper();
wrapper.find('.row.segment').forEach(comp => comp.set({ type: 'sektion' }));

But its not changing the types.

**Reply by artf** [reply]
You should be able to update, save and reload the project. You can't update the component view type at runtime once it's rendered, the alternative would be to remove the current component and add a new one with another type.
